lib/common/xml.c

```diff
@@ -669,13 +669,36 @@ create_xml_node(xmlNode * parent, const char *name)
     return node;
 }
 
+/*!
+ * \internal
+ * \brief Set a given string as an XML node's content
+ *
+ * \param[in,out] node     Node whose content to set
+ * \param[in]     content  String to set as the content
+ * \param[in]     escape   If \c true, escape special characters
+ *
+ * \note \c xmlNodeSetContent() does not escape special characters.
+ */
+void
```

Comment:
This could return a value since `xmlNodeSetContent()` returns `1` for generic error and `-1` for memory allocation failure. I kept it simple for now since the existing code doesn't do error-checking, but I could add it.